views/analysis.py

```diff
@@ -544,6 +547,23 @@ def validate_analysis_extent(request):
         return HttpResponseServerError()
 
 
+def calculate_area(request):
+    if request.method != 'POST':
+        return HttpResponseBadRequest()
+
+    if 'area' in request.session:
+        area = request.session['area']
```

Comment:
Oh, ok I get it why you put it on session.
This might not be good, because AnonymousUser shares the same session even though they maybe different users. Also, if you put it on session, the user will see the same area even if they want to see different impact layers.

Is it ok if you use validate_analysis_extent to get the area?
In waiting confirmation you can get the hazard_id, exposure_id, and analysis_extent, so you can make the same request.